### ISSUE TITLE
Make Pod 6 A Separated Stuff

### ISSUE THREAD

**Issue by tisonkun** [rakudo/rakudo#1893]
Right now we parse pod 6 along with perl 6 code, is it possible to make document stuff separated? That is, like `javadoc` for java.

Anyway, Perl 6 can reconcile many slangs, but why not make pod 6 a more independent module? We now generate document by EVAL file and use `$=pod` variable, which potentially executes unexpected code.

**Comment by tbrowder** [comment]
I have been thinking more about pod, slang, etc., and I agree with you. I don’t know why pod has to be completely parsed as part of the Perl 6 compiler. I think just skipping over it should be good enough as it is treated like whitespace or comments.

**Comment by lizmat** [comment]
It needs to be parsed as part of the code, otherwise we wouldn’t have any introspection on the $= variables?

**Comment by niner** [comment]
Can it even be completely skipped over by the compiler? What about custom extensions to POD? Or DOC BEGIN, DOC INIT, etc. blocks?

**Comment by jnthn** [comment]
I don't think it can be skipped over like that, no. However, it can most probably be successfully extracted to a slang like the `Q` and `Regex` languages are, if that is considered desirable.

**Comment by jnthn** [comment]
> I don't think it can be skipped over like that, no.

To expand on this a bit: it's of course possible, but it looks to me like the amount of work you'd need to do to find "where is the end" rivals doing a full parse of the Pod 6 anyway. "One-pass parsing" is a pretty fundamental Perl 6 language design principle, so anything that diverges from that is liable to go badly. Extracting it to a slang would let us retain one-pass parsing, while also providing a way to just start parsing in the Pod 6 language, thus avoiding the security issues of a general `EVAL`.

**Comment by tbrowder** [comment]
Thanks, Jonathan, a voice of reason in the wilderness, as usual!

**Comment by tbrowder** [comment]
This issue has been raised again since the Github folks are now looking at supporting pod6 but are concerned about the security aspects of Perl 6 when parsing code in the wild.

**Comment by TisonKun** [comment]
closed as no longer interested. feel free to create a new issue for further discussion.

**Comment by AlexDaniel** [comment]
It's still an issue, in fact a big one.
